activerecord/test/cases/relation/select_test.rb

```diff
@@ -16,5 +16,10 @@ def test_reselect
       expected = Post.select(:title).to_sql
       assert_equal expected, Post.select(:title, :body).reselect(:title).to_sql
     end
+
+    def test_reselect_with_default_scope_select
+      expected = PostWithDefaultSelect.reselect(:title).to_sql
+      assert_equal expected, Post.select(:title).to_sql
```

Comment:
‘reselect’ is what we test, it should be put right hand side.

Reply:
Thanks, @kamipo . Fixed it.